build_test.go

```diff
@@ -0,0 +1,287 @@
+					{Name: "vsdbg"},
+				},
+			},
+			Platform: packit.Platform{Path: "platform"},
+			Layers:   packit.Layers{Path: layersDir},
+			Stack:    "some-stack",
+		}
+	})
+
+	it.After(func() {
+		Expect(os.RemoveAll(layersDir)).To(Succeed())
+		Expect(os.RemoveAll(cnbDir)).To(Succeed())
+		Expect(os.RemoveAll(workingDir)).To(Succeed())
+
+	})
+
+	it("returns a result that installs vsdbg", func() {
+		result, err := build(buildContext)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(result.Layers).To(HaveLen(1))
+		layer := result.Layers[0]
+
+		Expect(layer.Name).To(Equal("vsdbg"))
+
+		Expect(layer.Path).To(Equal(filepath.Join(layersDir, "vsdbg")))
+
+		Expect(layer.SharedEnv).To(HaveLen(2))
+		Expect(layer.SharedEnv["PATH.delim"]).To(Equal(":"))
+		Expect(layer.SharedEnv["PATH.append"]).To(Equal(filepath.Join(layersDir, "vsdbg")))
+
+		Expect(layer.BuildEnv).To(BeEmpty())
+		Expect(layer.LaunchEnv).To(BeEmpty())
+		Expect(layer.ProcessLaunchEnv).To(BeEmpty())
+
+		Expect(layer.Build).To(BeFalse())
+		Expect(layer.Launch).To(BeFalse())
+		Expect(layer.Cache).To(BeFalse())
+
+		Expect(layer.Metadata).To(HaveLen(1))
+		Expect(layer.Metadata["dependency_sha"]).To(Equal("vsdbg-dependency-sha"))
+
+		Expect(layer.SBOM.Formats()).To(Equal([]packit.SBOMFormat{
+			{
+				Extension: sbom.Format(sbom.CycloneDXFormat).Extension(),
+				Content:   sbom.NewFormattedReader(sbom.SBOM{}, sbom.CycloneDXFormat),
+			},
+			{
+				Extension: sbom.Format(sbom.SPDXFormat).Extension(),
+				Content:   sbom.NewFormattedReader(sbom.SBOM{}, sbom.SPDXFormat),
+			},
+		}))
+
+		Expect(dependencyManager.ResolveCall.Receives.Path).To(Equal(filepath.Join(cnbDir, "buildpack.toml")))
+		Expect(dependencyManager.ResolveCall.Receives.Id).To(Equal("vsdbg"))
+		Expect(dependencyManager.ResolveCall.Receives.Version).To(Equal(""))
+		Expect(dependencyManager.ResolveCall.Receives.Stack).To(Equal("some-stack"))
+
+		Expect(dependencyManager.DeliverCall.Receives.Dependency).To(Equal(postal.Dependency{
+			ID:      "vsdbg",
+			Name:    "vsdbg-dependency-name",
+			SHA256:  "vsdbg-dependency-sha",
+			Stacks:  []string{"some-stack"},
+			URI:     "vsdbg-dependency-uri",
+			Version: "vsdbg-dependency-version",
+		}))
+
+		Expect(dependencyManager.DeliverCall.Receives.CnbPath).To(Equal(cnbDir))
+		Expect(dependencyManager.DeliverCall.Receives.DestinationPath).To(Equal(filepath.Join(layersDir, "vsdbg")))
+		Expect(dependencyManager.DeliverCall.Receives.PlatformPath).To(Equal("platform"))
+
+		Expect(sbomGenerator.GenerateCall.Receives.Dir).To(Equal(filepath.Join(layersDir, "vsdbg")))
+
+		Expect(buffer.String()).To(ContainSubstring("Some Buildpack some-version"))
+		Expect(buffer.String()).To(ContainSubstring("Executing build process"))
+		Expect(buffer.String()).To(ContainSubstring("Installing Visual Studio Debugger"))
+
+		info, err := os.Stat(filepath.Join(layersDir, "vsdbg", "vsdbg"))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(info.Mode().String()).To(Equal("-rwx--x---"))
```

Comment:
It looks to me like this bit is asserting that `Deliver()` has been called. If that's the only goal, it's probably easier to assert on `dependencyManagerDeliverCall.CallCount`.  Is there something else this assertion is meant to cover as well?

Reply:
The intention was to test whether the permissions eventually end up as owner and group-executable but this is poorly written on my part. I'll update.

Reply:
Ah, that makes sense. I missed the detail about the difference in the group-execute permissions in the test. I'm curious - what makes it necessary to chmod the executable when we haven't had to do that before? And if this is likely to come up for other dependencies/other buildpacks, do you think we need to change `dependency.Deliver()` in packit?

Reply:
From what I have seen, most of our other dependencies (definitely for cpython, go, node and ruby) are extracted into their layers with their binaries' execute permissions already set and located in a `bin` subdirectory which the lifecycle adds to the `PATH` variable by default. Neither of these is true in the case of `vsdbg`. I suppose we could modify deliver to perhaps check for a bin directory and certain permissions, but I think it might make more sense to let the buildpack handle that since it knows best the kind of file structure/permissions it would need to use its dependency.

Reply:
That makes sense. Maybe we revisit this once we're further along in our federated dependency journey. As new patterns emerge, we can push things into packit.